src/Symfony/Component/Notifier/Bridge/OneSignal/OneSignalTransport.php

```diff
@@ -0,0 +1,115 @@
+    {
+        $this->appId = $appId;
+        $this->apiKey = $apiKey;
+        $this->defaultRecipientId = $defaultRecipientId;
+
+        parent::__construct($client, $dispatcher);
+    }
+
+    public function __toString(): string
+    {
+        if (null === $this->defaultRecipientId) {
+            return sprintf('onesignal://%s@%s', urlencode($this->appId), $this->getEndpoint());
+        }
+
+        return sprintf('onesignal://%s@%s?recipientId=%s', urlencode($this->appId), $this->getEndpoint(), $this->defaultRecipientId);
+    }
+
+    public function supports(MessageInterface $message): bool
+    {
+        return $message instanceof PushMessage && (null !== $this->defaultRecipientId || ($message->getOptions() instanceof OneSignalOptions && null !== $message->getOptions()->getRecipientId()));
+    }
+
+    /**
+     * @see https://documentation.onesignal.com/reference/create-notification
+     */
+    protected function doSend(MessageInterface $message): SentMessage
+    {
+        if (!$message instanceof PushMessage) {
+            throw new UnsupportedMessageTypeException(__CLASS__, PushMessage::class, $message);
+        }
+
+        if ($message->getOptions() && !$message->getOptions() instanceof OneSignalOptions) {
+            throw new LogicException(sprintf('The "%s" transport only supports instances of "%s" for options.', __CLASS__, OneSignalOptions::class));
+        }
+
+        if (!($opts = $message->getOptions()) && $notification = $message->getNotification()) {
+            $opts = OneSignalOptions::fromNotification($notification);
+        }
+
+        $recipientId = $message->getRecipientId() ?? $this->defaultRecipientId;
+
+        if (null === $recipientId) {
+            throw new LogicException(sprintf('The "%s" transport should have configured `defaultRecipientId` via DSN or provided with message options.', __CLASS__));
+        }
+
+        $options = $opts ? $opts->toArray() : [];
+        $options['app_id'] = $this->appId;
+        $options['include_player_ids'] = [$recipientId];
+
+        if (!isset($options['headings'])) {
+            $options['headings'] = ['en' => $message->getSubject()];
+        }
+        if (!isset($options['contents'])) {
+            $options['contents'] = ['en' => $message->getContent()];
+        }
+
+        $response = $this->client->request('POST', 'https://'.$this->getEndpoint().'/api/v1/notifications', [
+            'headers' => [
+                'Accept' => 'application/json',
+                'Authorization' => 'Basic '.$this->apiKey,
+            ],
+            'json' => $options,
+        ]);
+
+        if (200 !== $response->getStatusCode()) {
+            throw new TransportException(sprintf('Unable to send the OneSignal push notification: "%s".', $response->getContent(false)), $response);
+        }
+
+        $result = $response->toArray(false);
```

Comment:
This can throw a decoder exception

Reply:
Not sure what you want me to do here :)

Reply:
Please wrap it in try/catch like other bridges. 
Say that the http body cannot be parsed into an array. That will throw an exception. Even though you pass false as first argument.

Reply:
I didn't found which other transports does the same what you ask, that's why I didn't understand :)

Reply:
Hm. I was sure I made a few prs to wrap toArray in a try/catch. 

Maybe it was reverted during the reviews and I forgotten. 

As long as we are doing the same thing across bridges then I’m happy. Feel free to ignore this.